echidna/scripts/dump_shrink.py

```diff
@@ -0,0 +1,100 @@
+""" Example shrinking script
+
+This script:
+  * Reads in spectrum from hdf5
+  * shrinks spectrum with user input pars and low high values
+  * Saves shrunk spectrum to given output or, by default, to the same
+    directory with _shrink added to the filename.
+
+Examples:
+  To shrink hdf5 file ``example.hdf5`` that has dimensions energy_mc and
+  radial3_mc to a range in energy of 1. to 4. MeV and radial3 from 0. to 0.5
+  and save the spectrum to ``example2.hdf5`` then the following command is
+  required::
+
+    $ python dump_shrink.py -i /path/to/example.hdf5 -o /path/to/example2.hdf5
+      -p energy_mc radial3_mc -l 1. 0. -u 4. 0.5
+
+  This will create the shrunk hdf5 file ``/path/to/example2.hdf5``.
+
+"""
+
+from echidna.output import store
+import echidna.utilities as utilities
+import os
+
+_logger = utilities.start_logging()
+
+
+def main(args):
+    """Smears energy and dumps spectra.
+
+    Args:
+      args (Namespace): Container for arguments. See
+        >>> python dump_smeared_energy.py -h
+    """
+    if not args.input:
+        parser.print_help()
+        raise ValueError("No input file provided")
+    if not args.pars:
+        parser.print_help()
+        raise ValueError("No parameters provided")
+    if not args.low and not args.up:
+        parser.print_help()
+        raise ValueError("Must provide lower and/or upper bounds to"
+                         "shrink to.")
+    spectrum = store.load(args.input)
+    num_pars = len(args.pars)
+    if args.low and args.up:
+        if len(args.low) != num_pars or len(args.up) != num_pars:
+            raise ValueError("Must have the same number of pars as bounds")
```

Comment:
@mjmottram, is this an instance where it would be better to try warn the user that there is a mismatch between the number of bounds and number of parameters, but try to run anyway?